src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -265,6 +282,7 @@ private struct RouteEntry
         public IEnumerable<string>? HttpMethods { get; init; }
         public RouteAttributes RouteAttributes { get; init; }
         public ThrowOnAddAfterEndpointBuiltConventionCollection Conventions { get; init; }
+        public List<Action<EndpointBuilder>> FinalConventions { get; init; }
```

Comment:
We should probably be as strict with throwing for normal conventions and final conventions.